src/lib.rs

```diff
@@ -155,6 +157,40 @@ pub fn build_from_docker(
     Ok((file_output, measurements))
 }
 
+/// Creates new enclave name
+///
+/// Requests the names of all running instances and checks the
+/// occurrence of the chosen name for the new enclave.
+pub fn new_enclave_name(run_args: RunEnclavesArgs) -> NitroCliResult<String> {
+    let enclave_name = match run_args.enclave_name {
+        Some(enclave_name) => enclave_name,
+        None => {
+            // Get name of EIF file from path eg. path/to/eif/hello.eif -> hello
+            let path_split: Vec<&str> = run_args.eif_path.split(&['/', '.'][..]).collect();
+            path_split[path_split.len() - 2].to_string()
+        }
+    };
+
+    let mut names = get_all_enclave_names().map_err(|e| {
+        e.add_subaction("Failed to handle all enclave process replies".to_string())
+            .set_action("Get Enclaves Name".to_string())
+    })?;
+
+    let mut idx = 0;
+    let mut result_name = enclave_name.clone();
+    names.sort();
+
+    // If duplicates are found, add index to name eg. testName -> testName_1 -> testName_2 ..
+    for name in names {
+        if result_name == name {
```

Comment:
Once an enclave is assigned a name with the index, it will be added to this array. I realised that it is not guaranteed that I will find all the existing names with indexes in consecutive order in the sorted array (eg. list is: name, nameother, name_1 ...) so I should check this in another way.